src/main/kotlin/org/rust/ide/intentions/import/ImportNameIntention.kt

```diff
@@ -130,14 +133,95 @@ class ImportNameIntention : RsElementBaseIntentionAction<ImportNameIntention.Con
         return if (shouldBePublicMods.all { it.isPublic }) return importInfo else null
     }
 
-    private fun RsPath.canBeResolvedToSuitableItem(project: Project, context: RsMod, info: ImportInfo): Boolean {
-        val externCrateName = if (info is ImportInfo.ExternCrateImportInfo && !info.target.isStd) info.target.normName else null
+    private fun RsPath.canBeResolvedToSuitableItem(
+        project: Project,
+        context: RsMod,
+        info: ImportInfo,
+        attributes: RsFile.Attributes
+    ): Boolean {
+        val externCrateName = if (info !is ImportInfo.ExternCrateImportInfo ||
+            attributes == RsFile.Attributes.NONE && info.target.isStd ||
+            attributes == RsFile.Attributes.NO_STD && info.target.isCore) {
+            null
+        } else {
+            info.target.normName
+        }
         val path = RsCodeFragmentFactory(project)
             .createPathInTmpMod(context, this, info.usePath, externCrateName) ?: return false
         val element = path.reference.resolve() ?: return false
         return !(element.parent is RsMembers && element.ancestorStrict<RsTraitItem>() != null)
     }
 
+    private fun filterForSingleItem(
+        candidates: List<ImportCandidate>,
+        fileAttributes: RsFile.Attributes
+    ): List<ImportCandidate> {
+        val candidatesWithPackage = mutableListOf<Pair<ImportCandidate, CargoWorkspace.Package>>()
+
+        val stdlibCandidates = mutableListOf<Pair<ImportCandidate, CargoWorkspace.Package>>()
+
+        for (candidate in candidates) {
+            val pkg = candidate.importItem.containingCargoTarget?.pkg ?: continue
+            val container = if (pkg.origin == PackageOrigin.STDLIB) stdlibCandidates else candidatesWithPackage
+            container += candidate to pkg
+        }
+
+        candidatesWithPackage += filterStdlibCandidates(stdlibCandidates, fileAttributes)
+        val pkgToCandidates = candidatesWithPackage.groupBy({ (_, pkg) -> pkg }, { (candidate, _) -> candidate })
+        return pkgToCandidates.flatMapTo(ArrayList()) { (_, candidates) -> filterInPackage(candidates) }
```

Comment:
`ArrayList()`?

Reply:
Hm, it's an artifact from some intermediate version.
Of course, just `flatMap` need here